### PR TITLE
feat: introducing shots as a primitive

### PR DESCRIPTION
**Pull request by monsieurBoutte**
# Overview

This branch adds a new `shots` primitive for Mux assets. It exposes helpers to request shot detection, fetch current shot results, and wait for completion, then wires that primitive into the public exports, docs, and test suite.

# What was changed

- Added `src/primitives/shots.ts` with three public APIs:
  - `requestShotsForAsset()` to start shot generation
  - `getShotsForAsset()` to read pending/completed shot data
  - `waitForShotsForAsset()` to optionally create and then poll until completion
- Exported the new primitive from `src/primitives/index.ts`.
- Added unit coverage for response transformation, request paths, polling behavior, timeout handling, and the "already requested" API error case.
- Added an integration test that waits for shots on a real test asset and validates the returned shot metadata.
- Updated `docs/PRIMITIVES.md` to document the new APIs and show how shot data can be used alongside transcripts and storyboards.

# Suggested review order

1. Read `src/primitives/shots.ts` to understand the new API surface and polling/error-handling behavior.
2. Read `tests/unit/shots.test.ts` to verify the intended contract and edge cases.
3. Read `tests/integration/shots.test.ts` to confirm the real-world workflow against a Mux asset.
4. Skim `src/primitives/index.ts` and `docs/PRIMITIVES.md` for public-surface/export and documentation updates.

---

> [!NOTE]
> **Medium Risk**
> Introduces new network calls and polling logic against a Mux Video API endpoint, which could impact reliability and runtime behavior (timeouts/retries) if the endpoint contract or error shapes differ.
> 
> **Overview**
> Adds a new `shots` primitive to request and retrieve Mux asset shot-detection results, including a `waitForShotsForAsset` helper that can auto-request, poll with bounded retries, and surface terminal `errored`/timeout states.
> 
> Wires the primitive into the public exports and updates `docs/PRIMITIVES.md` with API docs and an example combining shots with transcripts/storyboards. Adds unit tests for response shaping, request paths, polling/edge cases (min poll interval, already-requested handling, timeouts) and an integration test that runs shot detection against a real test asset.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 291f5be23e7401c5da74d1f42d4d41b34cfb697d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/primitives/shots.ts [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+ */
+export async function requestShotsForAsset(
+  assetId: string,
+  options: ShotRequestOptions = {},
+): Promise<PendingShotsResult> {
+  "use step";
+  const { credentials } = options;
+  const muxClient = await getMuxClientFromEnv(credentials);
+  const mux = await muxClient.createClient();
+  const response = await mux.post<Record<string, never>, ShotLocationsApiResponse>(
```
Comment (walkerfrankenberg) [inline src/primitives/shots.ts]: should we define the type here?

File: src/primitives/shots.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+      await requestShotsForAsset(assetId, { credentials });
+    } catch (error) {
+      if (!isShotsAlreadyRequestedError(error)) {
+        throw error;
+      }
+    }
+  }
+
+  const normalizedMaxAttempts = Math.max(1, maxAttempts);
+  const normalizedPollIntervalMs = Math.max(0, pollIntervalMs);
```
Comment (walkerfrankenberg) [inline src/primitives/shots.ts]: are we worried at all about a poll interval of 0 firing off a ton of API requests and getting limited or blowing up our system?

**Review by walkerfrankenberg — APPROVED** [review]
One comment on the poll interval minimum but otherwise lgtm

File: src/primitives/shots.ts (hunk 2 above)
Comment (monsieurBoutte) [reply src/primitives/shots.ts]: yup, good spot. I put up a fix to guard against hot loops
